### PR TITLE
Concatenated some truncated lines for easier translation

### PR DESCRIPTION
**Pull request by DisasterMo**
## Description

Some text in RetroArch UI is 'pieced together', sometimes word for word, from different hashes. This makes it rather difficult to translate said text. Furthermore, many symbols and punctuation are as of yet 'hardcoded' and not localizable, even though they might not fit all languages the same way.

In this PR I fixed up the following:

- "Missing", "Present",  "Optional" & "Required" have been grouped up accordingly (including commas and colons)
- A number of texts using MENU_ENUM_LABEL_VALUE_PORT ("Port") now have their own hashes (mostly to allow localisation of symbols/punctuation)
- Properly concatenated "Closed", "Ejected" & "Failed to"+"eject"/"close" with "virtual disc tray" to full sentences
- Concatenated "Auto-loading save state from" with "failed"/"succeeded"
- ... as well as "Internal Storage Status" & "Read/Write"/"Read only"

**FULL DISCLOSURE:**
While the code compiles for me, and I could verify the proper functionality for some strings, I could not verify all (especially the on-screen messages).
This PR does 'break' the (non-English) localisations somewhat, but that's only temporary, since the corresponding files will be overwritten by the crowdin translations, anyway.

## Related Issues
#10990 (related, but not resolved entirely yet)

## Related Pull Requests
This is inspired by #11096.

### CODE REVIEW

**Comment by twinaphex** [comment]
Why is "to" now hardcoded?

**Comment by michaelburgardt** [comment]
For 2 reasons:

1. It is used for warnings and log messages only, and as stated above, those confuse me (most of those are mostly hardcoded).
2. It is one of the overzealous, unnecessary changes, most of which I reverted back before pushing. This one I simply missed.

Anyway, while we're here, maybe you can answer my question: Should RARCH_WARN and RARCH_LOG messages be translatable, or not?

**Comment by michaelburgardt** [comment]
Oh, ah, @hunterk was of a different opinion, thus my most recent commit. But having a single word like "to" as a hash really doesn't seem like a good idea. Does the user see RARCH_WARN messages? I mean most users aren't likely to see the log, and trouble-shooting will likely be easier with a purely English log file, won't it?

**Comment by michaelburgardt** [comment]
Okay, @twinaphex, I know you must have like a billion other things on your mind, and this might not be the best place to discuss such things, but I do need some guidelines to work by. The way those messages are handled as of now is very inconsistent.

So, here's my suggestion:
Since both RARCH_WARN and RARCH_LOG use the same functions to write their messages to a file (not sure what else might be happening, but hey, that's something), I suggest to treat them the same way:
- Always write a (hard-coded) English message (so help can be provided regardless of user language)
- If RetroArch is set to use a language other than English, also write a translated message (I know there is a variable that can be probed for this, but I will likely need some time and help to figure out how to do this). So, in case the user changes the localisation, 2 lines will be written for each warning/log message (if no translation available this probably can be reduced to 1 line via str compare).

While I wait for a response, I'll just go back and construct a new "clean" PR for the other strings.

**Comment by twinaphex** [comment]
Looks like there's quite some merge conflicts you'll have to resolve first.

@guoyunhe Can you review this entire PR and leave behind your thoughts?

**Comment by michaelburgardt** [comment]
> Looks like there's quite some merge conflicts you'll have to resolve first.

On it.

File: intl/msg_hash_us.h
Comment (guoyunhe) [inline intl/msg_hash_us.h:10757]: Maybe "Virtual disc tray is closed." is more natural?

**Review by guoyunhe — COMMENTED** [review]
I think it is a good idea to not join translated strings, because some language is right-to-left (like Arabian). Can you provide some screenshots of these changes so we can see the actual visual result?

File: intl/msg_hash_us.h
Comment (michaelburgardt) [reply intl/msg_hash_us.h:10757]: Adjusting the actual phrasing is not the purpose of this PR. Not to mention, that this is a log message describing what happened, thus I think the current phrasing is fitting as is.

**Comment by michaelburgardt** [comment]
> I think it is a good idea to not join translated strings, because some language is right-to-left (like Arabian). Can you provide some screenshots of these changes so we can see the actual visual result?

Thank you for your response, @guoyunhe. But I am a little confused. The purpose of this PR is not to join "translated strings", but the English ones.

Currently, translators on crowdin have to deal with strings like:
"virtual disc tray.",
"Failed to" and
"close"
on their own, without any hint that those strings have anything to do with each other (outside of some comments by @IlDucci).

In code these strings are clumped together into sentences ("Failed to" + "close" + "virtual disc tray.") in fixed order and that can be problematic for other languages with different grammar.
For example, in German translating these strings in isolation can lead to very bad sentences, with mistakes like having verbs where one would need a noun & wrong conjugation.

To make it easier for translators these instances should be converted to full translatable sentences ("Failed to close virtual disc tray."). Doing this requires modifying the non-US translation header files (otherwise the code will usually not compile), but those will be replaced by the new crowdin translations anyway.

Since the purpose of this PR is to just construct the full sentences that these strings are formed into anyway, RetroArch shouldn't (and in the instances I could verify: doesn't) look any different (in English, anyway; I am _not_ reconstructing these sentences for other languages manually, that's what we have crowdin for).

**Anyway**:
The **actual** question here is, how log/warning/error messages should be handled: Many of those are hard-coded (i.e. untranslatable English only), some are completely translatable (like the example sentence above), and a few have both translatable as well as hard-coded components:

RARCH_WARN("[SRAM]: SRAM is larger than implementation expects, "    **<- hard-coded**
               "doing partial load (truncating %u %s %s %u).\n",                           **<-**
               (unsigned)rc,
               msg_hash_to_str(MSG_BYTES),    **<- translatable**
               msg_hash_to_str(MSG_TO),           **<-**
               (unsigned)mem_info.size);

File: intl/msg_hash_us.h
Comment (IlDucci) [reply intl/msg_hash_us.h:10757]: This log message also appears as a notification, so I do not know.

File: intl/msg_hash_us.h
Comment (IlDucci) [inline intl/msg_hash_us.h:11957]: This string does appear, mostly in consoled-based versions of RetroArch.
PAL60

Comment (michaelburgardt) [reply intl/msg_hash_us.h:11957]: Really? But I can't find the hash (MENU_ENUM_LABEL_VALUE_PAL60_ENABLE) being used anywhere in the code. Weird.
Well, shows what I know, I guess.

Comment (IlDucci) [reply intl/msg_hash_us.h:11957]: This screenshot was made a few months ago, so I do not know if this option was actually deleted. You should get a PS3 daily and see if it still has this.

Comment (michaelburgardt) [reply intl/msg_hash_us.h:11957]: Uh, do I need a PS3 to check that? Or how else can I do it?

Comment (IlDucci) [reply intl/msg_hash_us.h:11957]: Not really. RPCS3 can emulate the UI.

Comment (michaelburgardt) [reply intl/msg_hash_us.h:11957]: Ah yes, I forgot that was an option. Anyway, after finally getting it to run I can confirm that the option in question is no longer there:
Screenshot from 2021-03-13 23-28-16

**Comment by IlDucci** [comment]
I'm sorry I took so long to check it out by myself instead of asking, but I rechecked on my own and the option is still there (using the xbiso compile of the PS3 release). You must have the Europe Console Region enabled in the emulator's System tab in order to activate PAL60 support.

**Comment by DisasterMo** [comment]
> You must have the Europe Console Region enabled in the emulator's System tab in order to activate PAL60 support.

Still, I cannot reproduce your result. I used a v1.9.0 release of RetroArch for PS3 from Jan 04, 2021 ([link](https://store.brewology.com/ahomebrew.php?brewid=152)). If you can tell me where I can find this "xbiso compile of the PS3 release", I might have a look.
But honestly, it's just a FIXME that is not even relevant to the PR. I'll just remove them.

**Comment by IlDucci** [comment]
Whoops! Sorry, I said the wrong site. It's here where I am talking about, and again, I forgot to add that as a comment to the PAL60 line, and not to the main convo: https://xbins.org/libretro/nightly/playstation/ps3/

**Comment by DisasterMo** [comment]
Okay, @twinaphex, I addressed all voiced concerns (except for @guoyunhe suggestion, but again: I disagree with it and changing the messages is not the purpose of this PR, anyway).

Is there anything else, or can this be merged now?

**Comment by twinaphex** [comment]
There's still a merge conflict.

**Comment by twinaphex** [comment]
@guoyunhe After the merge conflict is resolved, do you feel content enough to go along with what is proposed here, and can this be merged? Or are there blocking issues you insist on seeing addressed first before that can happen?

**Comment by DisasterMo** [comment]
So, @twinaphex, what do I need to do here to get this PR merged?

**Comment by DisasterMo** [comment]
@twinaphex  I have addressed all voiced concerns. Can this be merged now?

**Comment by guoyunhe** [comment]
I am back online for a moment :-)

From Crowdin part, here is no blocker to modify source English strings, many or few. It does bring some work to translators to review and update affected translations. But Crowdin will work as normal. When you merge the PR, all strings go to Crowdin and our community members will start working on it.

From English language part, I am not a good English speaker. So I don't have an opinion. Ask some native English speakers for suggestions.

This is all my contribution to this PR...

**Comment by DisasterMo** [comment]
Thank you, @guoyunhe!
I can assure both you and @twinaphex that many changes in this PR have been requested by other translators as well. While merging this PR will add some work to the pile, it will ultimately make it easier to provide proper & correct translations for the affected texts. And upholding the principles of this PR in the future (I guess I should add them to the docs) will result in less headache for the translators overall.

**Comment by DisasterMo** [comment]
@jdgleaver I was instructed to request your review on this.
It's a rather simple & small PR; it's purpose is to do away with the 'broken up' sentences, which are a real headache to translate properly.
I was not happy with the multiple 'Port' related strings I introduced before, so instead I removed the hash (#) symbol from them, as there is no requirement I can see to use 'Port #%u' over 'Port %u'.
(For a bit more context: the hash is not universally used as a number/enumeration symbol, and thus would otherwise need to be localised for some languages -> no hash, no problem)

**Review by jdgleaver — APPROVED** [review]

**Comment by jdgleaver** [comment]
@DisasterMo This all looks fine to me. Seems like a good change.

**Comment by DisasterMo** [comment]
@jdgleaver Thank you for your approval!
PR has been squashed & rebased, and is ready for merge, @twinaphex

**Pull request merged**
